internal/fetcher/image_config/image_config.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2023 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package image_config
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+)
+
+// Fetcher retrieves the config for an image from its OCI registry.
+type Fetcher struct{}
```

Comment:
Doesn't have state or implement an interface, do we need the struct?

Reply:
We do not. At some point, this patch included having a list of fetchers that the caller would just iterate through. It made sense then. Removed for now. We'll likely come up with some interface at some point, but it is premature to do that right now.